handlers/messagebird/messagebird.go

```diff
@@ -0,0 +1,271 @@
+		}
+	}
+
+	jsonBody, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, sendUrl, bytes.NewReader(jsonBody))
+
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	var bearer = "AccessKey " + authToken
+	req.Header.Set("Authorization", bearer)
+
+	resp, _, err := handlers.RequestHTTP(req, clog)
+	if err != nil || resp.StatusCode/100 != 2 {
+		return status, nil
+	}
+
+	status.SetStatus(courier.MsgWired)
+
+	return status, nil
+}
+
+func verifyToken(tokenString string, secret string) (jwt.MapClaims, error) {
+	// Parse the token with the provided secret to get the claims
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		// Validate the signing method
+		// We only allow HS256
+		// ref: https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		// Return the secret used to sign the token
+		return []byte(secret), nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the token is valid
+	if token.Valid {
+		tokenClaims := token.Claims.(jwt.MapClaims)
+		return tokenClaims, nil
+	}
+
+	return nil, fmt.Errorf("Invalid token or missing payload_hash claim")
+}
+
+func calculateSignature(body []byte) string {
+	preHashSignature := sha256.Sum256(body)
+	return hex.EncodeToString(preHashSignature[:])
+}
+
+func (h *handler) validateSignature(c courier.Channel, r *http.Request) error {
+	if !h.validateSignatures {
+		return nil
+	}
+	headerSignature := r.Header.Get(signatureHeader)
+	if headerSignature == "" {
+		return fmt.Errorf("missing request signature")
+	}
+	configsecret := c.StringConfigForKey(courier.ConfigSecret, "")
+	if configsecret == "" {
+		return fmt.Errorf("missing configsecret")
+	}
+	verifiedToken, err := verifyToken(headerSignature, configsecret)
+	if err != nil {
+		return err
+	}
+	payloadHash := verifiedToken["payload_hash"].(string)
+
+	body, err := handlers.ReadBody(r, maxRequestBodyBytes)
+	if err != nil {
+		return fmt.Errorf("unable to read request body: %s", err)
+	}
+
+	expectedSignature := calculateSignature(body)
+	if !hmac.Equal([]byte(expectedSignature), []byte(payloadHash)) {
+		return fmt.Errorf("invalid request signature, signature doesn't match expected signature for body.")
+	}
+	return nil
+}
+
+type Message struct {
+	Recipients []string `json:"recipients"`
+	Originator string   `json:"originator"`
+	Subject    string   `json:"subject,omitempty"`
+	Body       string   `json:"body,omitempty"`
+	MediaURLs  []string `json:"mediaUrls,omitempty"`
+}
+
+type ReceivedMessage struct {
+	Receiver          string    `json:"receiver"`
+	Sender            string    `json:"sender"`
+	Message           string    `json:"message"`
+	Date              int       `json:"date"`
```

Comment:
The fields we do not really use here can just be removed entirely such as `Date`, `DateUTC` since we will save `CreatedDatetime` only

same for other we do not need can be removed.